Classes/Views/LabelListView.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  LabelListView.swift
+//  Freetime
+//
+//  Created by Joe Rocca on 12/6/17.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+final class LabelListView: UIView, UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+    
+    static func height(width: CGFloat, labels: [RepositoryLabel], cacheKey: String) -> CGFloat {
+        if let cachedHeight = LabelListViewHeightCache.shared.height(forKey: cacheKey) {
```

Comment:
One crucial issue: the height is only cached based on the given key. What happens when you change orientations? I think internally here we should just merge the **width and key** into a shared key and it will work?

```swift
let key = "\(width)\(cacheKey)"
let cachedHeight = LabelListViewHeightCache.shared.height(forKey: key)
```

Something like that.

Also, can we just define the cache within this class? The cache object itself is very simple.

```swift
private static let cache = [String: CGFloat]()
```

Reply:
```swift
let height = LabelListView.height(width: width, labels: labels, cacheKey: cacheKey + "\(width)")
```
I am actually doing this like above, but internally is way better!
Sounds good with the cache!